lib/samples/change_camera_controller/change_camera_controller.dart

```diff
@@ -0,0 +1,233 @@
+                      hintText: 'Select a camera controller',
+                      onSelected: handleCameraControllerSelection,
+                      initialSelection: _selectedCameraControllerKind,
+                    ),
+                  ],
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Future<void> onSceneViewReady() async {
+    // Add the scene to the view controller.
+    final scene = _setupScene();
+    _sceneViewController.arcGISScene = scene;
+
+    // Graphics overlay for the plane
+    final graphicsOverlay = GraphicsOverlay();
+    _planeGraphic = await _setupPlaneGraphic();
+    graphicsOverlay.graphics.add(_planeGraphic!);
+    graphicsOverlay.sceneProperties.surfacePlacement =
+        SurfacePlacement.absolute;
+    _sceneViewController.graphicsOverlays.add(graphicsOverlay);
+
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  // Function to handle when the user selects a camera controller from the
+  // dropdown menu.
+  void handleCameraControllerSelection(
+    CameraControllerKind? cameraControllerKind,
+  ) {
+    if (cameraControllerKind == null || _planeGraphic == null) return;
+
+    // Set the appropriate camera controller based on the user's selection.
+    switch (cameraControllerKind) {
+      case CameraControllerKind.globe:
+        _sceneViewController.cameraController = GlobeCameraController();
+      case CameraControllerKind.orbitGeoElement:
+        final cameraController = OrbitGeoElementCameraController(
+          targetGeoElement: _planeGraphic!,
+          distance: 1700,
+        );
+
+        cameraController.cameraPitchOffset = 3;
+        cameraController.cameraHeadingOffset = 150;
+        _sceneViewController.cameraController = cameraController;
+      case CameraControllerKind.orbitLocation:
+        final cameraController =
+            OrbitLocationCameraController.withTargetPositionAndCameraDistance(
+              targetLocation: ArcGISPoint(
+                x: -109.929589,
+                y: 38.437304,
+                z: 1700,
+                spatialReference: SpatialReference.wgs84,
+              ),
+              distance: 10000,
+            );
+
+        cameraController.cameraPitchOffset = 3;
+        cameraController.cameraHeadingOffset = 150;
+        _sceneViewController.cameraController = cameraController;
+    }
+
+    setState(() {
+      _selectedCameraControllerKind = cameraControllerKind;
+    });
+  }
+
+  ArcGISScene _setupScene() {
+    // Create a scene with an imagery basemap style.
+    final scene = ArcGISScene.withBasemapStyle(BasemapStyle.arcGISImagery);
+
+    // Set the scene's initial viewpoint.
+    scene.initialViewpoint = Viewpoint.withPointScaleCamera(
```

Comment:
This is more of an observation and thinking aloud - would become a separate issue as part of verifying the scenes API rather than a direct reflection on this sample PR. 

I went round in circles on the `Viewpoint` constructors that handle `Camera`. They seemed wordy but then actually required I think as there are a lot of ways of constructing a viewpoint. That said, are there are any logical opportunities to combine any constructors there?

Reply:
The verboseness is definitely helpful here, just making sure we revisit and review as we go. Thanks for your thoughts.